### DISCUSSION TITLE
[Feature Request] Ability to pass metadata when calling an action

### DISCUSSION THREAD

**Discussion by AndreMaz** [next-safe-action/next-safe-action#239] [Ideas]
Hey @TheEdoRan 

Would it be possible to add the ability to pass metadata to the action handler? Something like:
```typescript
import { fetchSomeData } from "@/app/fetchSomeDataHandler";

const data = {id}
const metadata = { skipAuth: true }
const res = await fetchSomeData(data, metadata)
```

Then, make `metadata` accessible in `actionClient` middleware

**Comment by TheEdoRan** [comment]
Hi, metadata is intended for server side stuff, so it's currently server side only and production applications already rely on that behavior. Furthermore, that implementation would conflict with [bound arguments](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#passing-additional-arguments). I think it's better to keep it this way and pass data to the server via bound arguments/main argument.
